Sources/MuxPlayerSwift/FairPlay/FairPlaySessionManager.swift

```diff
@@ -187,38 +193,53 @@ class DefaultFairPlayStreamingSessionManager<
                 self.logger.debug(
                     "Applicate certificate request failed with error: \(error.localizedDescription)"
                 )
+                let error = FairPlaySessionError.because(cause: error)
                 requestCompletion(Result.failure(
-                    FairPlaySessionError.because(cause: error)
+                    error
                 ))
+                self.errorDispatcher.dispatchApplicationCertificateRequestError(
+                    error: error,
+                    playbackID: playbackID
+                )
                 return
             }
             // error case: I/O finished with non-successful response
             guard responseCode == 200 else {
                 self.logger.debug(
                     "Applicate certificate request failed with response code: \(String(describing: responseCode))"
                 )
+                let error = FairPlaySessionError.httpFailed(
```

Comment:
For clarity, it looks like the errors dispatched to mux data will only give the type of request (App Cert vs. License) and the particular status, but no additional context, correct? This was different from what I thought you would be doing from prior convos, which was trying to get granular and actionable messages sent to Mux Data, a la e.g. https://github.com/muxinc/elements/pull/965/files#diff-6bc54449b29c68761250a41d4d58fd58052f5d932f195952c66908324b1fe436R98-R175

This will still be an improvement, but doesn't quite align with what I thought you were working towards (unless I'm missing something).

Reply:
Assuming the more granular messaging will not be resolved in this iteration. Going to simply mark as resolved.